### PR TITLE
add troubleshooting section on blocking RNGs

### PR DESCRIPTION
**Pull request by taccatisid**
Signed-off-by: Taccat Isid <taccatisid@protonmail.com>

## Pull request checklist

Use the following list to make sure your PR fits the Besu doc quality standard.

### Before creating the pull request

Make sure that:

- [x] [all commits in this PR are signed off for the DCO](https://wiki.hyperledger.org/display/BESU/DCO).
- [x] you read the [contribution guidelines](https://wiki.hyperledger.org/display/BESU/Contributing+to+documentation).
- [x] you have [tested your changes locally](https://wiki.hyperledger.org/display/BESU/MkDocs+And+Markdown+Guide#MkDocsAndMarkdownGuide-PreviewTheDocumentation) before submitting them to the community for review.

### After creating your pull request and tests finished

Make sure that:

- [x] you fixed all the issues raised by the tests, if any.
- [x] you verified the rendering of your changes on [ReadTheDocs.org PR preview](https://wiki.hyperledger.org/display/BESU/MkDocs+And+Markdown+Guide#MkDocsAndMarkdownGuide-PreviewwithReadTheDocs)
  and updated the testing link (see [Testing](#testing)).

## Describe the change

This adds a trouble-shooting section for users experiencing a blocked thread due to a misconfigured randomness source as in https://github.com/hyperledger/besu/issues/3013

### For content changes

- [x] Doc content
- [ ] Doc pages organisation

## Testing

https://hyperledger-besu--867.org.readthedocs.build/en/867/HowTo/Troubleshoot/Troubleshooting/#thread-blocked-due-to-lack-of-entropy-in-the-system-random-number-generator

### CODE REVIEW

File: docs/HowTo/Troubleshoot/Troubleshooting.md [hunk 1]
```diff
@@ -134,3 +134,38 @@ Reset taps to point to the correct remote branches by running `brew tap --repair
...
+        at java.base@11.0.11/java.io.FilterInputStream.read(FilterInputStream.java:133)
+        at java.base@11.0.11/sun.security.provider.NativePRNG$RandomIO.readFully(NativePRNG.java:424)
+        at java.base@11.0.11/sun.security.provider.NativePRNG$RandomIO.implGenerateSeed(NativePRNG.java:441)
+        at java.base@11.0.11/sun.security.provider.NativePRNG.engineGenerateSeed(NativePRNG.java:226)
+        ...
+```
+
+If this happens the Besu node can become unresponsive.
+
+The occurence of this problem and the possible solutions are system
```
Comment (macfarla) [inline docs/HowTo/Troubleshoot/Troubleshooting.md:159]: s/occurence/occurrence

File: docs/HowTo/Troubleshoot/Troubleshooting.md [hunk 2]
```diff
@@ -134,3 +134,38 @@ Reset taps to point to the correct remote branches by running `brew tap --repair
...
+        at java.base@11.0.11/java.io.FileInputStream.readBytes(Native Method)
+        at java.base@11.0.11/java.io.FileInputStream.read(FileInputStream.java:279)
+        at java.base@11.0.11/java.io.FilterInputStream.read(FilterInputStream.java:133)
+        at java.base@11.0.11/sun.security.provider.NativePRNG$RandomIO.readFully(NativePRNG.java:424)
+        at java.base@11.0.11/sun.security.provider.NativePRNG$RandomIO.implGenerateSeed(NativePRNG.java:441)
+        at java.base@11.0.11/sun.security.provider.NativePRNG.engineGenerateSeed(NativePRNG.java:226)
+        ...
+```
+
+If this happens the Besu node can become unresponsive.
```
Comment (macfarla) [inline docs/HowTo/Troubleshoot/Troubleshooting.md:157]: If this happens,

File: docs/HowTo/Troubleshoot/Troubleshooting.md [hunk 3]
```diff
@@ -134,3 +134,38 @@ Reset taps to point to the correct remote branches by running `brew tap --repair
...
+on Linux, using virtual machines without randomness source, and very
+early after computer startup.
+
+See [Entropy gathering in
+Java](https://www.baeldung.com/java-security-egd) to learn how to
+circumvent this issue. On Linux a quick, non-persistent workaround
+would be to run
+
+```bash
+sudo mount /dev/urandom /dev/random -o bind
```
Comment (macfarla) [inline docs/HowTo/Troubleshoot/Troubleshooting.md:170]: do we need to explain why this helps ie blocking vs non-blocking entropy?

File: docs/HowTo/Troubleshoot/Troubleshooting.md [hunk 4]
```diff
@@ -134,3 +134,38 @@ Reset taps to point to the correct remote branches by running `brew tap --repair
...
+        at java.base@11.0.11/sun.security.provider.NativePRNG.engineGenerateSeed(NativePRNG.java:226)
+        ...
+```
+
+If this happens the Besu node can become unresponsive.
+
+The occurence of this problem and the possible solutions are system
+dependent. The issue itself is rare, but would be more likely to occur
+on Linux, using virtual machines without randomness source, and very
+early after computer startup.
```
Comment (macfarla) [inline docs/HowTo/Troubleshoot/Troubleshooting.md:162]: maybe 3 bullet points here to make it clear they are three separate conditions?

**Comment by taccatisid** [comment]
The link to https://www.baeldung.com/java-security-egd is behind a very aggressive cloudflare ddos protection that more often than not fails the link checker. It always works when using a browser (with enabled JS). Is there a common workaround such problems?

**Comment by taccatisid** [comment]
> The link to https://www.baeldung.com/java-security-egd is behind a very aggressive cloudflare ddos protection that more often than not fails the link checker. It always works when using a browser (with enabled JS). Is there a common workaround such problems?

Changed the link to a different website.

**Review by alexandratran — APPROVED** [review]

**Pull request merged**